Minh/taskNavigationComponent_AndroidJetPack/app/src/main/java/com/example/demo/ui/call/CallViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.demo.ui.call
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.demo.data.model.SuggestedSearches
+import com.example.demo.data.repository.MetaDataReposirotyImpl
+import com.example.demo.data.repository.MetaDataRepository
+import com.example.demo.datasource.remote.RemoteDataSource
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class CallViewModel : ViewModel() {
+    private var _data=MutableLiveData<List<SuggestedSearches>>()
+    val data:LiveData<List<SuggestedSearches>> get() = _data
+    private val dataRepository:MetaDataRepository=MetaDataReposirotyImpl(RemoteDataSource())
+
+    fun getData(){
+        viewModelScope.launch (Dispatchers.IO){
```

Comment:
Tạo một extension riêng để xử lý connect APi 

package com.japanrelief.mylibrary.network.error

import retrofit2.HttpException
import java.io.IOException
import java.net.SocketTimeoutException

/**
 * Trace exceptions(api call or parse data or connection errors) &
 * depending on what exception returns [ApiError]
 *
 * */
fun traceErrorException(throwable: Throwable?): ApiError {

    return when (throwable) {

        is HttpException -> {
            when (throwable.code()) {
                400 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.BAD_REQUEST
                )
                401 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.UNAUTHORIZED
                )
                403 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.FORBIDDEN
                )
                404 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.NOT_FOUND
                )
                405 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.METHOD_NOT_ALLOWED
                )
                409 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.CONFLICT
                )
                500 -> ApiError(
                    throwable.message(),
                    throwable.code(),
                    ApiError.ErrorStatus.INTERNAL_SERVER_ERROR
                )
                else -> ApiError(
                    UNKNOWN_ERROR_MESSAGE,
                    0,
                    ApiError.ErrorStatus.UNKNOWN_ERROR
                )
            }
        }

        is SocketTimeoutException -> {
            ApiError(throwable.message, ApiError.ErrorStatus.TIMEOUT)
        }

        is IOException -> {
            ApiError(throwable.message, ApiError.ErrorStatus.NO_CONNECTION)
        }

        else -> ApiError(UNKNOWN_ERROR_MESSAGE, 0, ApiError.ErrorStatus.UNKNOWN_ERROR)
    }
}